pkg/services/ngalert/ngalert.go

```diff
@@ -34,15 +34,19 @@ const (
 	// because this could cause existing alert definition
 	// with intervals that are not exactly divided by this number
 	// not to be evaluated
-	baseIntervalSeconds = 10
-	// default alert definiiton interval
-	defaultIntervalSeconds int64 = 6 * baseIntervalSeconds
+	defaultBaseIntervalSeconds = 10
+	// default alert definition interval
+	defaultIntervalSeconds int64 = 6 * defaultBaseIntervalSeconds
 )
 
 func ProvideService(cfg *setting.Cfg, dataSourceCache datasources.CacheService, routeRegister routing.RouteRegister,
 	sqlStore *sqlstore.SQLStore, dataService *tsdb.Service, dataProxy *datasourceproxy.DataSourceProxyService,
-	quotaService *quota.QuotaService, backgroundServices *backgroundsvcs.Container) (*AlertNG, error) {
-	baseInterval := baseIntervalSeconds * time.Second
+	quotaService *quota.QuotaService, backgroundServices *backgroundsvcs.Container, m *metrics.Metrics) (*AlertNG, error) {
+	baseInterval := cfg.AlertingBaseIntervalSeconds
```

Comment:
The reason we didn't use a configuration here was in case there was a configuration (eg. 10 seconds) and then this value would change to something else (eg 15 seconds) then some alert rules created before the change can result not to be evaluated (rule intervals should be exactly divided by this scheduler interval).
So if we want to use a configuration we need to add more validations through out the code.

Reply:
@papagian it's only internal configuration though, and not public, so that tests can control this configuration. Is that a problem?

Reply:
Sorry, I didn't notice it. In that case it's not a problem!

Reply:
@papagian OK great :) I suppose I can add a comment to the configuration parameter, to flag it as internal.

Reply:
Added a comment flagging the parameter as internal.